app/charts/shared/chart-dimensions.tsx

```diff
@@ -211,3 +226,60 @@ export const useAxisLabelHeightOffset = ({
     labelWidth,
   };
 };
+
+export const getLongestXLabel = ({
+  xScale,
+  getXLabel,
+  xTimeUnit,
+  formatDate,
+  fontSize,
+}: {
+  xScale: ScaleBand<string>;
+  getXLabel: (d: any) => string;
+  xTimeUnit?: string;
+  formatDate?: (d: any, timeUnit: string) => string;
+  fontSize: number;
+}) => {
+  const domain = xScale.domain();
+
+  const formattedLabels = domain.map((d: any) => {
+    if (xTimeUnit && formatDate) {
+      return formatDate(d, xTimeUnit);
+    } else {
+      return getXLabel(d);
+    }
+  });
+
+  const labelWidths = formattedLabels.map((text: string) =>
+    getTextWidth(text, { fontSize })
+  );
+
+  const longestLabel = labelWidths.reduce((prev, current) =>
+    current > prev ? current : prev
+  );
+
+  return longestLabel;
+};
+
+const AXIS_TITLE_PADDING = 20;
+
+export const useXAxisTitleOffset = () => {
+  const [state] = useConfiguratorState(hasChartConfigs);
+  const chartConfig = getChartConfig(state);
+
+  const { xScale, getXLabel, xTimeUnit } = useChartState() as ColumnsState;
+  const { axisLabelFontSize } = useChartTheme();
+
+  return useMemo(() => {
+    return (
+      (isColumnConfig(chartConfig)
+        ? getLongestXLabel({
+            xScale,
+            getXLabel,
+            xTimeUnit,
+            fontSize: axisLabelFontSize,
+          })
+        : axisLabelFontSize * LINE_HEIGHT) + AXIS_TITLE_PADDING
+    );
+  }, [chartConfig, axisLabelFontSize, xScale, getXLabel, xTimeUnit]);
```

Comment:
Nit: I'd extract `chartConfig` out of the hook to not re-calculate when some non-related to X axis options change.